Promise.reject("Kaboom!").catch(
  (error: unknown) => console.error(error)
);
Promise.reject("Kaboom!").catch(
  function (error: unknown) { console.error(error); }
);

Promise.reject("Kaboom!").then(
  () => {},
  (error: unknown) => console.error(error)
);
Promise.reject("Kaboom!").then(
  function () {},
  function (error: unknown) { console.error(error); }
);

Promise.reject("Kaboom!").catch(
  (error: unknown) => console.error(error)
);
Promise.reject("Kaboom!").catch(
  function (error: unknown) { console.error(error); }
);

Promise.reject("Kaboom!").catch(
  (error: any) => console.error(error)
);
Promise.reject("Kaboom!").catch(
  function (error: any) { console.error(error); }
);

Promise.reject("Kaboom!").then(
  () => {},
  (error: unknown) => console.error(error)
);
Promise.reject("Kaboom!").then(
  function () {},
  function (error: unknown) { console.error(error); }
);

Promise.reject("Kaboom!").then(
  () => {},
  (error: any) => console.error(error)
);
Promise.reject("Kaboom!").then(
  function () {},
  function (error: any) { console.error(error); }
);

